### PR TITLE
feat(console): submit cloud questionnaire

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

- submit the cloud questionnaire
- redirect to the welcome page when the questionnaire is not submitted
- extract useMeCustomData hook to manage admin user custom data
- refactor the `useUserPerferences` hook with the `useMeCustomData` hook
- add an `useUserOnboardingData` to manage user onboarding data

## Testing

Test locally.

### CODE REVIEW

File: packages/console/src/cloud/hooks/use-user-cloud-data.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const useUserCloudData = () => {
+  const { data, error, isLoading, isLoaded, update: updateMeCustomData } = useMeCustomData();
+
+  const parseData = useCallback((): UserCloudData => {
+    try {
+      return z.object({ [key]: userCloudDataGuard }).parse(data).cloud;
+    } catch {
+      return {};
+    }
+  }, [data]);
```
Comment (simeng-li) [inline packages/console/src/cloud/hooks/use-user-cloud-data.ts:20]: zod.safeParse

File: packages/console/src/cloud/hooks/use-user-cloud-data.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  const userCloudData = useMemo(() => parseData(), [parseData]);
+
+  const update = useCallback(
+    async (data: Partial<UserCloudData>) => {
+      await updateMeCustomData({
+        [key]: {
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/console/src/cloud/hooks/use-user-cloud-data.ts:27]: Yes. The `adminConsolePreferences` also use this custom data, but it's in the `useUserPreferences` hook.

File: packages/console/src/cloud/hooks/use-user-cloud-data.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { useCallback, useMemo } from 'react';
+import { z } from 'zod';
+
+import useMeCustomData from '@/hooks/use-me-custom-data';
+
+import type { UserCloudData } from '../types';
+import { userCloudDataGuard } from '../types';
+
+const key = 'cloud';
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/console/src/cloud/hooks/use-user-cloud-data.ts]: Updated with constants.
Currently, 'cloud' is being used as a key in custom data, but if you have any better ideas for a name, please let me know.

File: packages/console/src/cloud/hooks/use-user-cloud-data.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/cloud/hooks/use-user-cloud-data.ts]: updated.

File: packages/console/src/cloud/hooks/use-user-cloud-data.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  const parseData = useCallback((): UserCloudData => {
+    try {
+      return z.object({ [key]: userCloudDataGuard }).parse(data).cloud;
+    } catch {
+      return {};
+    }
+  }, [data]);
+
+  const userCloudData = useMemo(() => parseData(), [parseData]);
+
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/console/src/cloud/hooks/use-user-cloud-data.ts]: updated.

File: packages/console/src/cloud/hooks/use-user-cloud-data.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/console/src/cloud/hooks/use-user-cloud-data.ts]: Rename to `onboarding`.

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**